### PR TITLE
meson: split files

### PR DESCRIPTION
**Pull request by yuwata**
To shorten the main meson.build file.

### CODE REVIEW

File: units/meson.build [hunk 1]
```diff
@@ -3,322 +3,673 @@
...
-        ['timers.target',                       ''],
-        ['tmp.mount',                           '',
-         'local-fs.target.wants/'],
-        ['umount.target',                       ''],
-        ['usb-gadget.target',                   ''],
-        ['user.slice',                          ''],
-        ['var-lib-machines.mount',              'ENABLE_MACHINED',
-         'remote-fs.target.wants/ machines.target.wants/'],
-        ['systemd-oomd.socket',                 'ENABLE_OOMD'],
+        { 'file' : 'basic.target' },
```
Comment (dtardon) [inline units/meson.build]: Isn't this a bit redundant? Wouldn't it be simpler to change the list to a dictionary with unit name as a key? I.e.,
```python
units = {
        'basic.target': {},
        ... 
        'cryptsetup-pre.target': { 'conditions': ['HAVE_LIBCRYPTSETUP'] },
        ...
}

Comment (dtardon) [reply units/meson.build]: Ah, it follows the same pattern as `executables`.

File: meson.build [hunk 2]
```diff
@@ -2421,1967 +2463,101 @@ test_dlopen = executable(
...
+        build = true
+        foreach cond : dict.get('conditions', [])
+                if conf.get(cond) != 1
+                        build = false
+                        break
                 endif
+        endforeach
+        if not build
+                continue
         endif
```
Comment (medhefgo) [inline meson.build:2685]: This  can go into the loop, no?

File: meson.build [hunk 3]
```diff
@@ -2425,1970 +2463,101 @@ test_dlopen = executable(
...
-
-        if want_tests != 'false'
-                test('test-systemd-tmpfiles',
-                     test_systemd_tmpfiles_py,
-                     # https://github.com/mesonbuild/meson/issues/2681
-                     args : exe.full_path())
+        endforeach
+        if not build
+                continue
         endif
```
Comment (medhefgo) [inline meson.build:2685]: Can go into the loop above.

File: meson.build [hunk 4]
```diff
@@ -2425,1970 +2463,101 @@ test_dlopen = executable(
...
-                public_programs += exe
-
-                if want_tests != 'false'
-                        test('test-systemd-tmpfiles.standalone',
-                             test_systemd_tmpfiles_py,
-                             # https://github.com/mesonbuild/meson/issues/2681
-                             args : exe.full_path())
+        kwargs = {}
+        foreach key, val : dict
+                if key == 'name' or key == 'dbus' or key == 'public' or key == 'conditions'
```
Comment (medhefgo) [inline meson.build]:
```suggestion
                if key in ['name', 'dbus', 'public', 'conditions']
```

**Review by medhefgo — COMMENTED** [review]
Oh boy. That last commit. :D But I love it.

Might be easier to digest/review if it were done in smaller batches. I also hope you sorted the unit list by name.

It's really silly meson doesn't provide functions.

I just take a quick peek at it and assume you made sure everything is in order.

**Comment by medhefgo** [comment]
This reminds me that I need to revive my meson features PR. But it be best to wait for this stuff to go in first or we might step on each others toes.

File: meson.build (hunk 2 above)
Comment (yuwata) [reply meson.build:2685]: Unfortunately, no. Moving `continue` in the internal loop make the internal one continued...

**Comment by yuwata** [comment]
> Might be easier to digest/review if it were done in smaller batches.

OK. I will.

**Pull request closed without merging**